### ISSUE TITLE
When loading C# dll with reference to google.proto System.Memory is not found

### ISSUE THREAD

**Issue by dudima123** [protocolbuffers/protobuf#9213]
System.TypeInitializationException: The type initializer for 'LoggerDescriptorProto' threw an exception. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Memory, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file specified.

Version: master/v3.19.3
Language: C# load by python clr package

Windows

Python 3.7.4

**What did you do?**
1. My .dll using protobuf as refrence version 3.19.3, there is an exception when trying to load it by clr python, older version as 3.7.0 works WO any problem

**Comment by chbeyg** [comment]
Any updates regarding this issue?

I get the same exception with:
Python - 3.10
Google.Protobuf - 3.19.4

**Comment by jskeet** [comment]
Yes, Google.Protobuf [now depends on System.Memory](https://www.nuget.org/packages/Google.Protobuf#dependencies-body-tab), and [version 3.7.0 didn't do so](https://www.nuget.org/packages/Google.Protobuf/3.7.0#dependencies-body-tab).

Fundamentally this isn't really an issue with the Google.Protobuf package - it's an issue with how you're loading the library. We assume that clients will supply the dependencies listed in the NuGet package. I don't know exactly how your Python code is loading Google.Protobuf, but I'd expect that if System.Memory is available to the CLR (perhaps being loaded explicitly first?) it would all be okay.

Although I believe System.Memory does have some "interesting" aspects in terms of its versioning, I suspect they're irrelevant, and that you'd have the same problem with *any* library with a dependency, unless you've satisfied that dependency first.

Are you able to try providing System.Memory.dll? If either you don't know how to do that or you try it and it doesn't work, could you provide a complete sample application that we could use to reproduce the problem? There are too many aspects that are unclear at the moment to help you beyond "make the dependency available".

**Comment by jskeet** [comment]
Closing as I don't believe there's anything for this library to do here.
